pkgs/development/compilers/rust/rustc.nix

```diff
@@ -105,8 +111,20 @@ stdenv.mkDerivation (finalAttrs: {
     ++ optional stdenv.hostPlatform.isDarwin "-rpath ${llvmSharedForHost.lib}/lib"
   );
 
-  # Increase codegen units to introduce parallelism within the compiler.
-  RUSTFLAGS = "-Ccodegen-units=10";
+  # We can't set RUSTFLAGS for the same target twice, so assume that if the
+  # rustc target name is the same, the flags are the same
+
+  "CARGO_TARGET_${stdenv.hostPlatform.rust.cargoEnvVarTarget}_RUSTFLAGS" =
+    rustFlagsFor stdenv.hostPlatform;
+
+  ${
+    if stdenv.hostPlatform.rust.cargoEnvVarTarget != stdenv.targetPlatform.rust.cargoEnvVarTarget then
+      "CARGO_TARGET_${stdenv.targetPlatform.rust.cargoEnvVarTarget}_RUSTFLAGS"
+    else
+      null
+  } =
+    rustFlagsFor stdenv.targetPlatform;
```

Comment:
If the `rust.framepointers` config option isn't working for us we should probably file a bug report rather than just working around it like this.

Reply:
wait, i remembered, i didn't do that because i didn't think it handles the cross case right, but i'll check for sure

Reply:
`rust.frame-pointers` does work, and integrates well with `fastCross`. I will change it to use that.